airbyte-cdk/python/airbyte_cdk/sources/declarative/types.py

```diff
@@ -16,43 +16,38 @@
 
 
 class Record(Mapping[str, Any]):
-    def __init__(self, data: dict, associated_slice: StreamSlice):
+    def __init__(self, data: Mapping[str, Any], associated_slice: Optional[StreamSlice]):
         self._data = data
         self._associated_slice = associated_slice
 
     @property
-    def data(self) -> dict:
+    def data(self) -> Mapping[str, Any]:
         return self._data
 
     @property
-    def associated_slice(self) -> StreamSlice:
+    def associated_slice(self) -> Optional[StreamSlice]:
         return self._associated_slice
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return repr(self._data)
 
-    def __setitem__(self, key: str, value: Any):
```

Comment:
This was used to support transformation before but as this is now really considered a "Mapping" (transformation occurs on the dict and not the Record), we can remove the set